gremlin-driver/src/test/java/org/apache/tinkerpop/gremlin/driver/WebSocketClientBehaviorIntegrateTest.java

```diff
@@ -271,4 +276,28 @@ public void shouldNotCreateReplacementConnectionWhenClientClosesConnection() thr
                         .filter(str -> str.contains("Considering new connection on"))
                         .count());
     }
-}
\ No newline at end of file
+
+    /**
+     * (TINKERPOP-2814) Tests to make sure that the SSL handshake is now capped by connectionSetupTimeoutMillis and not
+     * the default Netty SSL handshake timeout of 10,000ms.
+     */
+    @Test
+    public void shouldAttemptHandshakeForLongerThanDefaultNettySslHandshakeTimeout() {
+        final Cluster cluster = Cluster.build("localhost").port(SimpleSocketServer.PORT)
+                .minConnectionPoolSize(1)
+                .maxConnectionPoolSize(1)
+                .connectionSetupTimeoutMillis(20000) // needs to be larger than 10,000ms.
+                .enableSsl(true)
+                .create();
+
+        final Client.ClusteredClient client = cluster.connect();
+        final long start = System.currentTimeMillis();
+
+        try {
+            client.submit("1");
+        } finally {
+            // Test against 15,000ms which should give a big enough buffer to avoid timing issues.
```

Comment:
nit
not important but FYI, in some geographies (such as Germany) a comma in numbers is read as decimal and folks in Germany will probably read this as 15ms! Not using comma at all keeps things independent of such geographic irregularities.

Reply:
Thanks for the reminder. I'll remove those commas.